src/converters/degiroConverterV3.ts

```diff
@@ -160,33 +160,60 @@ export class DeGiroConverterV3 extends AbstractConverter {
           continue;
         }
 
-        // Look ahead in the remaining records if there is one with the samen orderId.
-        let matchingRecord = this.findMatchByOrderId(record, records.slice(idx + 1));
+        // Look ahead in the remaining records if there are some with the same orderId.
+        let matchingRecords = undefined;
+        if (record.orderId) {
+          matchingRecords = this.findMatchesByOrderId(record, deGiroRecords.slice(idx + 1));
+        }
+
+        // If there was no match by orderId, and there was no orderId present on the current record, look ahead one record to find a match by ISIN + Product + Date/Time.
+        if (!matchingRecords && !record.orderId) {
+          matchingRecords = this.findMatchByIsinSameDateTime(record, deGiroRecords.slice(idx + 1, idx + 2));
```

Comment:
I left out time previously because this can actually differ between records and then produce an unwanted result.

Reply:
Do you have examples? The previous logic did an unbounded search, which was too much. We may remove time (but keep date) or even remove date/time but then need to limit search window somehow (number of lines?)

Reply:
I recently fixed a bug, and the reporter had a dividend set which had a different time.

```
01-08-2024,07:40,31-07-2024,OXFORD SQUARE CAPITAL CORP,US69181V1070,Impôts sur dividende,,USD,-0.53,USD,2.97,
01-08-2024,07:39,31-07-2024,OXFORD SQUARE CAPITAL CORP,US69181V1070,Dividende,,USD,3.50,USD,3.50,
```

Reply:
> I recently fixed a bug, and the reporter had a dividend set which had a different time.
> 
> ```
> 01-08-2024,07:40,31-07-2024,OXFORD SQUARE CAPITAL CORP,US69181V1070,Impôts sur dividende,,USD,-0.53,USD,2.97,
> 01-08-2024,07:39,31-07-2024,OXFORD SQUARE CAPITAL CORP,US69181V1070,Dividende,,USD,3.50,USD,3.50,
> ```

I'd suggest to limit by date (not time) then